src/test/java/com/mate/fruitshop/service/impl/TransactionsProcessingServiceImplTest.java

```diff
@@ -0,0 +1,68 @@
+package com.mate.fruitshop.service.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.mate.fruitshop.db.Storage;
+import com.mate.fruitshop.model.FruitEntry;
+import com.mate.fruitshop.model.Transaction;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TransactionsProcessingServiceImplTest {
+    private static TransactionsProcessingServiceImpl processor =
+            new TransactionsProcessingServiceImpl();
+    private static List<Transaction> transactions;
+    private static List<FruitEntry> expectedStorage;
+
+    @Before
+    public void setUp() {
+        transactions = new ArrayList<>();
+        transactions.add(new Transaction(Transaction.Operation.BALANCE, "banana", 20));
+        transactions.add(new Transaction(Transaction.Operation.BALANCE, "apple", 100));
+        transactions.add(new Transaction(Transaction.Operation.SUPPLY, "banana", 100));
+        transactions.add(new Transaction(Transaction.Operation.PURCHASE, "banana", 13));
+        transactions.add(new Transaction(Transaction.Operation.RETURN, "apple", 10));
+        transactions.add(new Transaction(Transaction.Operation.PURCHASE, "apple", 20));
+        transactions.add(new Transaction(Transaction.Operation.PURCHASE, "banana", 5));
+        transactions.add(new Transaction(Transaction.Operation.SUPPLY, "banana", 50));
+        expectedStorage = new ArrayList<>();
+        expectedStorage.add(new FruitEntry("banana", 152));
+        expectedStorage.add(new FruitEntry("apple", 90));
+    }

```

Comment:
why do you initializate transaction before every before every test? do you really need it?

Reply:
for every test to be completely independent from each other